routing/missioncontrol.go

```diff
@@ -298,7 +235,115 @@ func generateBandwidthHints(sourceNode *channeldb.LightningNode,
 // if no payment attempts have been made.
 func (m *MissionControl) ResetHistory() {
 	m.Lock()
-	m.failedEdges = make(map[EdgeLocator]time.Time)
-	m.failedVertexes = make(map[route.Vertex]time.Time)
-	m.Unlock()
+	defer m.Unlock()
+
+	m.history = make(map[route.Vertex]*nodeHistory)
+
+	log.Debugf("Mission control history cleared")
+}
+
+// getEdgeProbability is expected to return the success probability of a payment
+// from fromNode along edge.
+func (m *MissionControl) getEdgeProbability(fromNode route.Vertex,
+	edge EdgeLocator, amt lnwire.MilliSatoshi) float64 {
+
+	m.Lock()
+	defer m.Unlock()
+
+	// Get the history for this node. If there is no history available,
+	// assume that it's success probability is 1. After the attempt new
+	// information becomes available to adjust this probability.
+	nodeHistory, ok := m.history[fromNode]
+	if !ok {
+		return 1
```

Comment:
should we use the default/configurable value her?

Reply:
Ok, this is done in later commit. Would be nice if the apriori value was defined before this, but not a blocker.

Reply:
Moved into this commit.